packages/block-library/src/navigation/index.php

```diff
@@ -76,8 +76,8 @@ function render_block_navigation( $attributes, $content, $block ) {
  */
 function build_navigation_html( $block, $colors ) {
 	$html = '';
-
-	$class_attribute = sprintf( ' class="%s"', esc_attr( $colors['css_classes'] ? 'wp-block-navigation-item__link ' . $colors['css_classes'] : 'wp-block-navigation-item__link' ) );
+	$css_classes = implode( ' ', $colors['css_classes'] );
```

Comment:
With these changes, we are already adding space between classes so we need to remove the space from ' has-%s-color' in:
$colors['css_classes'][] = sprintf( ' has-%s-color', $attributes['textColor'] );